pkg/workers/exec/konnector.go

```diff
@@ -79,12 +80,36 @@ type konnectorResult struct {
 
 // beforeHookKonnector skips jobs from trigger that are failing on certain
 // errors.
-func beforeHookKonnector(req *jobs.JobRequest) (bool, error) {
-	if req.Manual || req.Trigger == nil {
+func beforeHookKonnector(job *jobs.Job) (bool, error) {
+	konnectorMessage := struct {
+		Account      string `json:"account"`
+		Konnector    string `json:"konnector"`
+		FolderToSave string `json:"folder_to_save"`
+	}{}
+
+	if err := json.Unmarshal(job.Message, &konnectorMessage); err == nil {
+		inst, err := instance.Get(job.DomainName())
+		if err != nil {
+			return false, err
+		}
+		registries, _ := inst.Registries()
+		app, err := registry.GetApplication(konnectorMessage.Konnector, registries)
+		if err != nil {
+			job.Logger().Errorf("konnector %q could not get application to fetch maintenance status", konnectorMessage.Konnector)
+		} else if app.MaintenanceActivated {
+			if job.Manual && !app.MaintenanceOptions.FlagDisallowManualExec {
+				return true, nil
+			}
+			job.Logger().Infof("konnector %q has not been triggered because of its maintenance status", konnectorMessage.Konnector)
+			return false, nil
+		}
+	}
+
+	if job.Manual || job.TriggerID != "" {
 		return true, nil
 	}
-	trigger := req.Trigger
-	state, err := jobs.GetTriggerState(trigger)
+
+	state, err := jobs.GetTriggerState(job, job.TriggerID)
```

Comment:
Something looks wrong.

If `jobs.TriggerID` was not the empty string (`""`), the function would have already returned (`if job.Manual || job.TriggerID != "" { return true, nil }`). So, here , `jobs.TriggerID` must be the empty string, and `GetTriggerState` would fail.

Reply:
Fixed.